src/net/netsyscall.c

```diff
@@ -682,7 +682,9 @@ static sysreturn socket_write_udp(netsock s, void *source, u64 length,
     err_t err = ERR_OK;
 
     /* XXX check how much we can queue, maybe make udp bh */
-    /* XXX check if remote endpoint set? let LWIP check? */
+    if (!dest_addr && !udp_is_flag_set(s->info.udp.lw, UDP_FLAGS_CONNECTED))
```

Comment:
If it's possible for the write to race with a socket close then checking the udp.lw should be inside the lwip lock

Reply:
Agreed. I moved this check under the lwip lock.